packages/fast-usdc/src/pool-share-math.js

```diff
@@ -120,3 +125,66 @@ export const withFees = (shareWorth, fees) => {
   const balancePost = add(shareWorth.numerator, fees);
   return makeRatioFromAmounts(balancePost, shareWorth.denominator);
 };
+
+/**
+ *
+ * @param {Amount<'nat'>} requested
+ * @param {Amount<'nat'>} poolSeatAllocation
+ * @param {Amount<'nat'>} encumberedBalance
+ * @param {PoolStats} poolStats
+ * @throws {Error} if conditions are not met
+ */
+export const borrowCalc = (
+  requested,
+  poolSeatAllocation,
+  encumberedBalance,
+  poolStats,
+) => {
+  // pool must never go empty
+  !isGTE(requested, poolSeatAllocation) ||
+    Fail`Cannot borrow. Requested ${q(requested)} must be less than pool balance ${q(poolSeatAllocation)}.`;
+
+  return harden({
+    encumberedBalance: add(encumberedBalance, requested),
+    poolStats: {
+      ...poolStats,
+      totalBorrows: add(poolStats.totalBorrows, requested),
+    },
+  });
+};
+
+/**
+ * @param {ShareWorth} shareWorth
+ * @param {Allocation} fromSeatAllocation
+ * @param {RepayAmountKWR} amounts
+ * @param {Amount<'nat'>} encumberedBalance
+ * @param {PoolStats} poolStats
+ * @throws {Error} if conditions are not met
+ */
+export const repayCalc = (
+  shareWorth,
+  fromSeatAllocation,
+  amounts,
+  encumberedBalance,
+  poolStats,
+) => {
+  (isEqual(fromSeatAllocation.Principal, amounts.Principal) &&
+    isEqual(fromSeatAllocation.PoolFee, amounts.PoolFee) &&
+    isEqual(fromSeatAllocation.ContractFee, amounts.ContractFee)) ||
+    Fail`Cannot repay. From seat allocation ${q(fromSeatAllocation)} does not equal amounts ${q(amounts)}.`;
+
+  !isGTE(amounts.Principal, encumberedBalance) ||
+    isEqual(amounts.Principal, encumberedBalance) ||
+    Fail`Cannot repay. Principal ${q(amounts.Principal)} exceeds encumberedBalance ${q(encumberedBalance)}.`;
+
+  return harden({
+    shareWorth: withFees(shareWorth, amounts.PoolFee),
```

Comment:
maybe `withFees` should be inlined... or at least not exported any more.

Reply:
Agree, but there's an existing unit test that relies on it so I've left the export for now